smithy-model/src/main/java/software/amazon/smithy/model/traits/AnnotationTrait.java

```diff
@@ -15,31 +15,32 @@
 
 package software.amazon.smithy.model.traits;
 
+import java.util.Collections;
 import java.util.function.Function;
 import software.amazon.smithy.model.SourceException;
 import software.amazon.smithy.model.SourceLocation;
 import software.amazon.smithy.model.node.BooleanNode;
 import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.ObjectNode;
 import software.amazon.smithy.model.shapes.ShapeId;
 
 /**
- * Trait implementation that expects an empty object or a boolean
- * value of true.
+ * Trait implementation for traits that are an empty object.
  */
-public abstract class BooleanTrait extends AbstractTrait {
-    public BooleanTrait(ShapeId id, SourceLocation sourceLocation) {
+public abstract class AnnotationTrait extends AbstractTrait {
+    public AnnotationTrait(ShapeId id, SourceLocation sourceLocation) {
         super(id, sourceLocation);
     }
 
     @Override
     protected final Node createNode() {
-        return new BooleanNode(true, getSourceLocation());
+        return new ObjectNode(Collections.emptyMap(), getSourceLocation());
     }
 
     /**
      * Trait provider that expects a boolean value of true.
      */
-    public static class Provider<T extends BooleanTrait> extends AbstractTrait.Provider {
+    public static class Provider<T extends AnnotationTrait> extends AbstractTrait.Provider {
```

Comment:
Looks like there's now dead code in this class's `createTrait` method that should be cleaned up.